functions.py

```diff
@@ -1,84 +1,122 @@
 import json
 import os
+from keys import *
+from Command_factory import *
+from ICommand import ICommand
 
 
 def search(path, primary_key):
-    y = False
-    for r, d, f in os.walk(path):
-        if primary_key in f:
-            y = True
+    found = False
+    for roots, directories, files in os.walk(path):
+        if primary_key in files:
+            found = True
             break
-    return y
-
-
-def creates_dir(schema):
-    x = json.load(open(schema, 'r'))
-    p = os.path.join(os.getcwd(), x['database_name'])
-    os.mkdir(p)
-    for i in x['Tables']:
-        d = os.path.join(p, i['name'])
-        os.mkdir(d)
-
-
-def creates(schema, table, primary_key):
-    x = json.load(open(schema, 'r'))
-    path = os.getcwd() + '\\' + x['database_name'] + '\\' + table + '\\' + primary_key
-    f = open(path, 'w')
-    f.write("{\n")
-    i = 0
-    for k in x['Tables']:
-        if k['name'] != table:
-            i = i + 1
+    return found
+
+
+class create_dir_command(ICommand):
+    def __init__(self, schema):
+        self.schema = schema
+
+    def execute(self):
+        data = json.load(open(self.schema, 'r'))
+        path = os.path.join(os.getcwd(), data[keys().database])
+        os.makedirs(path, exist_ok=True)
+
+        for table in data[keys().Tables]:
+            t_path = os.path.join(path, table[keys().name])
+            os.makedirs(t_path, exist_ok=True)
+
+
+class create_command(ICommand):
+    def __init__(self, schema, table, primary_key):
+        self.schema = schema
+        self.table = table
+        self.primary_key = primary_key
+
+    def execute(self):
+        file = open(self.schema, 'r')
+        column = json.load(file)
+        file.close()
+        path = os.path.join(os.getcwd(), column[keys().database])
+        if not os.path.exists(path):
+            return False
+        path = path + '\\' + self.table + '\\' + self.primary_key
+        if not os.path.exists(path):
+            json_object = {}
+            file = open(path, 'w')
+            index = 0
+            for key in column[keys().Tables]:
+                if key[keys().name] != self.table:
+                    index = index + 1
+                else:
+                    column = column[keys().Tables][index]
+                    break
+            for element in column[keys().columns]:
+                json_object[element] = '0'
+            json.dump(json_object, file)

```

Comment:
create a private method called __write_table_schema. which is `schema["tables"][i]`